### PR TITLE
ASM-7477 Specs vnx_lun

### PR DESCRIPTION
**Pull request by hribab**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+end
+
+#gem "puppet", :path => File.dirname(__FILE__), :require => false
+gem "facter", *location_for(ENV['FACTER_LOCATION'] || '~> 1.6')
+gem "hiera", *location_for(ENV['HIERA_LOCATION'] || '~> 1.0')
+gem "rake", :require => false
+gem "rgen", "0.6.5", :require => false
+
+
+group(:development, :test) do
```
Comment (ripienaar) [inline Gemfile]: is all these really needed?

File: Gemfile.lock [hunk 2]
```diff
@@ -0,0 +1,50 @@
+GEM
```
Comment (gavin-scott) [inline Gemfile.lock]: I wouldn't commit this, usually that's only done for "applications".

File: spec/unit/provider/vnx_lun/vnx_lun_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  describe '#get_hlu_sg_of_lun' do
+    context 'When you pass lun number' do
+      it 'returns hlu and sg name of that LUN' do
+        provider.stubs(:get_lun_number).returns(3)
+        provider.stubs(:run).returns("Storage Group Name:    test_group\n  HLU Number     ALU Number\n  ----------     ----------\n  0               3")
+        expect(provider.get_hlu_sg_of_lun).to match_array([0, "test_group"])
+    end
+   end
+  end
+
```
Comment (gavin-scott) [inline spec/unit/provider/vnx_lun/vnx_lun_spec.rb]: no space here

File: spec/unit/provider/vnx_lun/vnx_lun_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      :ensure   => 'present',
+      :type     => 'nonthin',
+      :capacity => '20GB',
+      :pool_name=> 'Pool 0',
+    }
+    )}
+
+  let(:provider) {resource.provider}
+
+  describe '#size_to_sizequal' do
```
Comment (gavin-scott) [inline spec/unit/provider/vnx_lun/vnx_lun_spec.rb]: Please use `"` for all strings in new code. Esp. please be consistent, there's a mix of single and double quotes in this file.

File: spec/unit/provider/vnx_lun/vnx_lun_spec.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    context 'when you pass size' do
+      it 'returns capacity and units' do
+        expect(provider.size_to_sizequal("100GB")).to match_array([100, "gb"])
+        expect{provider.size_to_sizequal("gb100")}.to raise_error("gb100 is not a valid volume size.")
```
Comment (gavin-scott) [inline spec/unit/provider/vnx_lun/vnx_lun_spec.rb]: these two lines should be separate test cases.

**Comment by gavin-scott** [comment]
Can you add a `.travis.yml` to this commit so that Travis CI can run the unit tests with the PRs? See here for an example: https://github.com/dell-asm/dell-equallogic/blob/master/.travis.yml Since you don't have a Rakefile you can probably just change the script line to `"bundle exec rspec"`

File: .travis.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+language: ruby
```
Comment (gavin-scott) [inline .travis.yaml]: This needs to be named `.travis.yml` -- Travis CI is not finding this file due to the `.yaml` extension.

**Comment by gavin-scott** [comment]
@hribab  nice job getting the tests running. Please address my remaining comments and I'll get this merged.

**Pull request merged**